storage3/_async/file_api.py

```diff
@@ -435,18 +437,30 @@ async def _upload_or_update(
         """
         if file_options is None:
             file_options = {}
-        cache_control = file_options.get("cache-control")
+        cache_control = file_options.pop("cache-control", None)
```

Comment:
Confirming here that we don't want to keep the `cahce-control` and `upsert` headers